include/ddc/kernels/splines/matrix_corner_block.hpp

```diff
@@ -0,0 +1,328 @@
+        } else if (j >= m_nb) {
+            if constexpr (Kokkos::SpaceAccessibility<
+                                  Kokkos::DefaultHostExecutionSpace,
+                                  typename ExecSpace::memory_space>::accessible) {
+                m_Abm_1_gamma(i, j - m_nb) = aij;
+            } else {
+                // Inefficient, usage is strongly discouraged
+                Kokkos::deep_copy(
+                        Kokkos::subview(m_Abm_1_gamma, i, j - m_nb),
+                        Kokkos::View<const double, Kokkos::HostSpace>(&aij));
+            }
+        } else {
+            if constexpr (Kokkos::SpaceAccessibility<
+                                  Kokkos::DefaultHostExecutionSpace,
+                                  typename ExecSpace::memory_space>::accessible) {
+                m_lambda(i - m_nb, j) = aij;
+            } else {
+                // Inefficient, usage is strongly discouraged
+                Kokkos::deep_copy(
+                        Kokkos::subview(m_lambda, i - m_nb, j),
+                        Kokkos::View<const double, Kokkos::HostSpace>(&aij));
+            }
+        }
+    }
+    virtual void factorize() override
+    {
+        m_q_block->factorize();
+        m_q_block->solve_inplace(
+                detail::build_mdspan(m_Abm_1_gamma, std::make_index_sequence<2> {}));
+
+        calculate_delta_to_factorize();
+
+        m_delta->factorize();
+    }
+
+protected:
+    Matrix_Corner_Block(
+            int const n,
+            int const k,
+            std::unique_ptr<Matrix> q,
+            int const lambda_size1,
+            int const lambda_size2)
+        : Matrix(n)
+        , m_k(k)
+        , m_nb(n - k)
+        , m_q_block(std::move(q))
+        , m_delta(new Matrix_Dense<ExecSpace>(k))
+        , m_Abm_1_gamma("Abm_1_gamma", m_nb, m_k)
+        , m_lambda("lambda", lambda_size1, lambda_size2)
+    {
+        assert(n > 0);
+        assert(m_k >= 0);
+        assert(m_k <= n);
+        assert(m_nb == m_q_block->get_size());
+    }
+
+public:
+    virtual void calculate_delta_to_factorize()
+    {
+        auto delta_proxy = *m_delta;
+        auto lambda_host = Kokkos::
+                create_mirror_view_and_copy(Kokkos::DefaultHostExecutionSpace(), m_lambda);
+        auto Abm_1_gamma_host = Kokkos::
+                create_mirror_view_and_copy(Kokkos::DefaultHostExecutionSpace(), m_Abm_1_gamma);
+        Kokkos::parallel_for(
+                "calculate_delta_to_factorize",
+                Kokkos::MDRangePolicy<
+                        Kokkos::DefaultHostExecutionSpace,
+                        Kokkos::Rank<2>>({0, 0}, {m_k, m_k}),
+                [&](const int i, const int j) {
+                    double val = 0.0;
+                    for (int l = 0; l < m_nb; ++l) {
+                        val += lambda_host(i, l) * Abm_1_gamma_host(l, j);
+                    }
+                    delta_proxy.set_element(i, j, delta_proxy.get_element(i, j) - val);
```

Comment:
Can you comment on the thread safety of `set_element` ?

Reply:
Inlining of:
`delta_proxy.set_element(i, j, delta_proxy.get_element(i, j) - val);`

Is just:
'delta_proxy.m_a(i,j) = delta_proxy.m_a(i,j) - val;`

So I think there is no problem with it as we just act on a single element ?

Reply:
I agree now that we are not calling `deep_copy` anymore